practical-07/src/components/HomeView.vue

```diff
@@ -0,0 +1,225 @@
+      putData: "editData",
+      sendData: "addData",
+      deleteData: "deleteCarDetails",
+    }),
+  
+    detailDisplay(id) {
+      this.$router.push(`/car/${id}`);
+    },
+    deleteData(id) {
+     this.deleteData(id);
+    },
+    addCar(carItem) {
+      if (carItem.id !== "") {
+        this.carData = this.carData.map((car) => {
+          if (car.carId === carItem.id) {
+            car.carName = carItem.name;
+            car.carPrice = carItem.price;
+            car.carDetails = carItem.details;
+            car.carImage = carItem.image;
+          }
+          return car;
+        });
+        alert("Edited data  :\n" + JSON.stringify(carItem, 2, null));
+        this.putData(carItem);
+        this.resetInitialValues();     
+      } 
+      else {
+        carItem.id = new Date().getTime().toString(36);
+        this.sendData(carItem);
+        alert("Created data : \n" + JSON.stringify(carItem, 2, null));
+      }
+    },
+
+    editCarDetails(carId) {
+      this.formHeading = "Edit Car Details";
+      const carItem = this.carData.find((carItem) => carItem.id === carId);
+      this.initialValues = carItem;
+      this.showModel = true;
+    },
+    
+    handleModel(status) {
+      this.showModel = status;
+      if (status === false) {
+        this.formHeading = "";
+        this.resetInitialValues();
+      }
+    },
+
+    resetInitialValues() {
+      this.initialValues = {
+        carId: "",
+        carName: "",
+        carDetails: "",
+        carImage: "",
+        carPrice: undefined,
+      };
+    },
+  },
+}
+</script>
+
+<style>
+.column {
+  display: flex;
+  flex-direction: row;
+  float: left;
+  justify-content: center;
+  flex-wrap: wrap;
+}
+
+.add-btn {
+  float: right;
+  padding: 13px;
+  color: rgb(27, 75, 92);
+  background: rgb(149, 216, 249);
+  justify-content: center;
+  font-size: 14px;
+  border: none;
+  cursor: pointer;
+  border-radius: 8px;
+  margin-top: 55px;
+}
+.add-btn:hover {
+  background: rgb(27, 75, 92);
+  color: rgb(230, 230, 230);
+}
+
+@media screen and (max-width: 650px) {
+  .column {
+    width: 100%;
+    display: block;
+    margin-bottom: 10px;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    flex-wrap: wrap;
+  }
+}
+
+.dataContainer{
+ background-color: rgb(187, 234, 235);
+ border-radius: 15px;
+ margin: 25px;
+ width: 400px;
+ box-shadow: rgba(112, 112, 220, 0.1) 0px 8px 24px, rgba(223, 28, 28, 0.1) 0px 16px 56px, #f265651a 0px 24px 80px;
+ overflow: hidden;
+ position: relative;
+}
+
+button .text {
+  max-width: 0;
+  transition: max-width 0.5s;
+  display: inline-block;
+  vertical-align: top;
+  white-space: nowrap;
+  overflow: hidden;
+}
+
+button:hover .text {
+  max-width: 7rem;
+  transition-delay: 0.1s;
+}
+
+button .separator {
+  display: inline-block;
+  width: 0;
+  transition: width 0.1s;
+  transition-delay: 0.5s;
+}
+
+button:hover .separator {
+  width: 0.3em;
+  transition-delay: unset;
+}
+
+.list-move, 
```

Comment:
Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/mansinakrani/vuejs-practicals/pullRequest?prid=9375994)
```suggestion
.list-move,
```